streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueToTimestampedKeyValueByteStoreAdapter.java

```diff
@@ -127,14 +131,31 @@ public <R> QueryResult<R> query(
         final QueryConfig config) {
 
 
+
         final long start = config.isCollectExecutionInfo() ? System.nanoTime() : -1L;
-        final QueryResult<R> result = store.query(query, positionBound, config);
+        QueryResult<R> result = store.query(query, positionBound, config);
+        final Position position = result.getPosition();
+
+        if (result.isSuccess()) {
+            if (result.getResult() instanceof byte[]) {
```

Comment:
I think we should actually check the input `query` type instead:
```
if (query instanceof KeyQuery || query instanceof TimestampedKeyQuery)
```

Reply:
ok